### PR TITLE
Latest changes to prepare_release

### PR DESCRIPTION
**Pull request by gtiao**

### CODE REVIEW

File: helloworld.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+HELLO WORLD
```
Comment (konradjk) [inline helloworld.txt]: Hello!

File: variant_qc/correct_fafs.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                               .map(lambda x: hl.set(freq_ht.freq_meta[x].values())))
+    freq_ht = freq_ht.annotate(orig_faf95=hl.Struct(**make_faf_struct(freq_ht, faf_index_dict)))
+    freq_ht = freq_ht.annotate(faf=freq_ht.faf.map(lambda x: hl.cond(freq_ht.ac1_pops.contains(hl.set(x.meta.values())), x.annotate(faf95=0, faf99=0), x)))
+    freq_ht = freq_ht.annotate(new_faf95=hl.Struct(**make_faf_struct(freq_ht, faf_index_dict)))
+
+    # Check annotations
+    freq_ht.select('faf_ac', 'orig_faf95', 'new_faf95').show()
+
+    # Write
+    freq_ht.write(annotations_ht_path(data_type, 'frequencies_faf_corrected'))
```
Comment (konradjk) [inline variant_qc/correct_fafs.py]: I think this looks good, especially if you've already spot-checked. My only suggestion would be to drop the original ones before writing this (unless we only select freq and faf later on for the release, then it's probably fine). Your call.

File: helloworld.txt (hunk 1 above)
Comment (gtiao) [reply helloworld.txt]: Oops

File: variant_qc/correct_fafs.py (hunk 2 above)
Comment (gtiao) [reply variant_qc/correct_fafs.py]: Good point. I'll drop before writing here.

File: variant_qc/prepare_data_release.py [hunk 3]
```diff
@@ -817,6 +1006,8 @@ def main(args):
 
         # Remove gnomad_ prefix for VCF export
         ht = hl.read_table(release_ht_path(data_type, nested=False, temp=True))
+        contigs = hl.eval(ht.aggregate(hl.agg.collect_as_set(ht.locus.contig)))
```
Comment (konradjk) [inline variant_qc/prepare_data_release.py]: no hl.eval needed here

File: variant_qc/prepare_data_release.py [hunk 4]
```diff
@@ -19,6 +19,65 @@
 
 SORT_ORDER = ['popmax', 'group', 'pop', 'subpop', 'sex']
 
+POP_DICT = {
```
[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: Moved to `basics.py`

File: variant_qc/prepare_data_release.py [hunk 5]
```diff
@@ -19,6 +19,65 @@
...
+    'rf_train': {"Description": "Variant was used in training random forest model"},  # export?
+    'rf_tp_probability': {"Description": "Random forest prediction probability for a site being a true variant"},
+    'transmitted_singleton': {"Description": "Variant was a callset-wide doubleton that was transmitted within a family (i.e., a singleton amongst unrelated sampes in cohort)"},
+    'variant_type': {"Description": "Variant type (snv, indel, multi-snv, multi-indel, or mixed)"},
+    'allele_type': {"Number": "A", "Description": "Allele type (snv, ins, del, or mixed)"},
+    'n_alt_alleles': {"Number": "A", "Description": "Total number of alternate alleles observed at variant locus"},
+    'was_mixed': {"Description": "Variant type was mixed"},
+    'has_star': {"Description": "Variant locus coincides with a spanning deletion (represented by a star) observed elsewhere in the callset"},
+    'pab_max': {"Number": "A", "Description": "Maximum p-value over callset for binomial test of observed allele balance for a heterozygous genotype, given expectation of AB=0.5"},
+    'vep': {"Description": "Variant Effect Predictor annotations"}
```
[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: It can be... why, is it better that way?

[the remark this answers is not in the document]
Comment (konradjk) [reply variant_qc/prepare_data_release.py]: It looks like at some point later in the pipeline you do, bc the current release VCF has:
```
##INFO=<ID=vep,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_
position|Protein_position|Amino_acids|Codons|Existing_variation|ALLELE_NUM|DISTANCE|STRAND|FLAGS|VARIANT_CLASS|MINIMISED|SYMBOL_SOURCE|HGNC_ID|CANONICAL|TSL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|UNIPARC|GENE_PHEN
O|SIFT|PolyPhen|DOMAINS|HGVS_OFFSET|GMAF|AFR_MAF|AMR_MAF|EAS_MAF|EUR_MAF|SAS_MAF|AA_MAF|EA_MAF|ExAC_MAF|ExAC_Adj_MAF|ExAC_AFR_MAF|ExAC_AMR_MAF|ExAC_EAS_MAF|ExAC_FIN_MAF|ExAC_NFE_MAF|ExAC_OTH_MAF|ExAC_SAS_MAF|
CLIN_SIG|SOMATIC|PHENO|PUBMED|MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|LoF|LoF_filter|LoF_flags|LoF_info">
```

[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: Oh yes, it's ```        new_info_dict.update({'vep': {'Description': hl.eval(vep_csq_ht.globals.vep_csq_header)}})
```

[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: I'll just delete this item here

File: variant_qc/prepare_data_release.py [hunk 6]
```diff
@@ -187,6 +248,18 @@ def sample_sum_check(ht: hl.Table, prefix: str, label_groups: Dict[str, List[str
...
+    :param Table ht: Table containing annotations to be checked
+    :param Expression cond_expr: logical expression referring to annotations in ht to be checked
+    :param str check_description: String describing the condition being checked; is displayed in terminal summary message
+    :param list of str display_fields: List of names of ht annotations to be displayed in case of failure (for troubleshooting purposes);
+        these fields are also displayed if verbose is True
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks
+    :return: Terminal display of results from checking the sum of the specified annotations
+    :rtype: None
+    '''
```
[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: That's a great idea. The sanity check function *is* slow. I'll put it on my list of things to do for v3.

File: variant_qc/prepare_data_release.py [hunk 7]
```diff
@@ -19,6 +19,65 @@
...
+    'transmitted_singleton': {"Description": "Variant was a callset-wide doubleton that was transmitted within a family (i.e., a singleton amongst unrelated sampes in cohort)"},
+    'variant_type': {"Description": "Variant type (snv, indel, multi-snv, multi-indel, or mixed)"},
+    'allele_type': {"Number": "A", "Description": "Allele type (snv, ins, del, or mixed)"},
+    'n_alt_alleles': {"Number": "A", "Description": "Total number of alternate alleles observed at variant locus"},
+    'was_mixed': {"Description": "Variant type was mixed"},
+    'has_star': {"Description": "Variant locus coincides with a spanning deletion (represented by a star) observed elsewhere in the callset"},
+    'pab_max': {"Number": "A", "Description": "Maximum p-value over callset for binomial test of observed allele balance for a heterozygous genotype, given expectation of AB=0.5"},
+    'vep': {"Description": "Variant Effect Predictor annotations"}
+}
+
```
[the remark this answers is not in the document]
Comment (gtiao) [reply variant_qc/prepare_data_release.py]: I dunno, but I'm willing to give it a try and find out...

**Pull request merged**